test/src/sdk-harness/test_projects/storage_map/src/main.sw

```diff
@@ -0,0 +1,281 @@
+contract;
+
+use std::storage::StorageMap;
+
+pub struct Struct {
+    x: u32,
+    y: b256,
+    z: b256,
+}
+
+pub enum Enum {
+    V1: b256,
+    V2: u64,
+    V3: b256,
+}
+
+storage {
+    map1: StorageMap<u64,
+    bool>, map2: StorageMap<u64,
+    u8>, map3: StorageMap<u64,
+    u16>, map4: StorageMap<u64,
+    u32>, map5: StorageMap<u64,
+    u64>, map6: StorageMap<u64,
+    (b256, u8, bool) >, map7: StorageMap<u64,
+    Struct>, map8: StorageMap<u64,
+    Enum>, map9: StorageMap<u64,
+    str[33]>, map10: StorageMap<u64,
+    [b256;
+    3]>, map11: StorageMap<bool,
+    u64>, map12: StorageMap<u8,
+    u64>, map13: StorageMap<u16,
+    u64>, map14: StorageMap<u32,
+    u64>, map15: StorageMap<(b256,
+    u8, bool), u64 >, map16: StorageMap<Struct,
+    u64>, map17: StorageMap<Enum,
+    u64>, map18: StorageMap<str[33],
+    u64>, map19: StorageMap<[b256;
+    3],
+    u64>, 
+}
+
+abi StorageMapTest {
+    fn init();
+
+    fn into_u64_to_bool(key: u64, value: bool);
```

Comment:
In my head a map is "key _to_ value" so I went with `insert_into_<type1>_to_<type2>_map`. Let me know if that's clear!

Reply:
Ah, I just go by "key, value pair" or similarly "key to value". I would have still gone with `insert_<type2>_into_<type1>` since I primarily go by the "pair" thought and thus I think of storing a value under a specific key